src/libstd/sync/mod.rs

```diff
@@ -42,3 +42,18 @@ mod condvar;
 mod mutex;
 mod once;
 mod rwlock;
+
+/// Save power or switch hyperthreads in a busy-wait spin-loop.  This
+/// function is deliberately more primitive than
+/// `std::thread::yield_now` and does not directly yield to the
+/// system's scheduler.  In some cases it might be useful to use a
+/// combination of both functions.
+#[inline]
+#[stable(feature = "rust1", since = "1.1.6")]
```

Comment:
Here's an example: https://github.com/rust-lang/rust/blob/master/src/libcore/convert.rs#L291

Pick some arbitrary (unique) feature name, and before we merge this we'll create a tracking issue to refer to.